src/components/storage/aws.ts

```diff
@@ -114,27 +91,30 @@ export class AwsEventLogStorage implements EventLogStorage {
     }
 
     async storeEvents(events: any) {
-        const key = 'events/events.csv'
-
-        const allEvent = await this._getObject({key, type: 'csv'})
-
-        events.data.forEach((event) => {
-            allEvent.push([event.time, events.id, event.other, event.type])
-        })
+        let storeEventSuccess = true
+
+        for(let event of events.data) {
+            const body = {
+                time: event.time,
+                id: events.id,
+                other: event.other,
+                type: event.type
+            }
 
-        let body;
+            const eventId = event.time + event.type
```

Comment:
We talked about this briefly, but how big is the chance that an event of the same type happens on the same millisecond? What about adding in a string version of the `event.other` as part of the event ID? TBH I'm not really sure what this though (adding typings for events would be great here!).

Reply:
The `event.type` is reserved and this is JSON, so I don't think so we have to use this to make the id. BTW the same event cannot occur at the same time for the same user.